mlflow/utils/rest_utils.py

```diff
@@ -42,7 +41,9 @@ def http_request(
     elif host_creds.token:
         auth_str = "Bearer %s" % host_creds.token
 
-    headers = dict(_DEFAULT_HEADERS)
+    from mlflow.tracking.request_header.registry import resolve_request_headers
+
+    headers = dict({**_DEFAULT_HEADERS, **resolve_request_headers()})
```

Comment:
Is there a way we can easily unit test the addition of these provider headers to a request?